### ISSUE TITLE
Not working on Linux with GunGame

### ISSUE THREAD

**Issue by andreiseverin** [tmp64/weaponmod#12]
Hello,

I've tried to install the mod on my server and the weapons work fine when added in the `weapon.ini` file. When I try to use them with the [GunGame plugin](https://forums.alliedmods.net/showthread.php?t=180714) they won't deploy (I can see the HUD but I don't have the weapon). 

The plugin was compiled with the `#define HLWPNMOD` line. 

I tested it on my Windows server and I had the same behavior in the beginning. When I stopped and restarted the server, the new weapons deployed and worked as they should. Tried that on Linux and it doesn't work.

```aghl.ru
"aghl.ru" is "1.10.2+master.6750a4c"
version
Protocol version 48
Exe version 1.1.2.2/Stdio (valve)
ReHLDS version: 3.13.0.788-dev
Build date: 07:36:33 Jul 12 2023 (3378)
Build from: https://github.com/dreamstalker/rehlds/commit/f955b07
meta list
Currently loaded plugins:
      description     stat pend  file                      vers                         src  load  unload
 [ 1] JK_Botti        RUN   -    jk_botti_mm_i386.so       v1.44beta1                   ini  Start ANY  
 [ 2] AMX Mod X       RUN   -    amxmodx_mm_i386.so        v1.10.0.5461                 ini  Start ANY  
 [ 3] Reunion         RUN   -    reunion_mm_i386.so        v0.1.0.137                   ini  Start Never
 [ 4] Rechecker       RUN   -    rechecker_mm_i386.so      v2.7                         ini  Chlvl ANY  
 [ 5] ReAuthCheck     RUN   -    reauthcheck_mm_i386.so    v0.1.6                       ini  Start Never
 [ 6] CURL AMXX       RUN   -    curl_amxx_i386.so         v1.1.1                       pl2  ANY   ANY  
 [ 7] Orpheu          RUN   -    orpheu_amxx_i386.so       v2.6.3                       pl2  ANY   ANY  
 [ 8] ReAPI           RUN   -    reapi_amxx_i386.so        v5.22.0.254-dev              pl2  ANY   Never
 [ 9] Engine          RUN   -    engine_amxx_i386.so       v1.10.0.5461                 pl2  ANY   ANY  
 [10] FakeMeta        RUN   -    fakemeta_amxx_i386.so     v1.10.0.5461                 pl2  ANY   ANY  
 [11] Ham Sandwich    RUN   -    hamsandwich_amxx_i386.so  v1.10.0.5461                 pl2  ANY   ANY  
 [12] Fun             RUN   -    fun_amxx_i386.so          v1.10.0.5461                 pl2  ANY   ANY  
 [13] HL Weapon Mod   RUN   -    weaponmod_amxx_i386.so    v0.1.0-dev.1+bc87b53.master  pl2  ANY   Never
 [14] Amxx Easy Http  RUN   -    easy_http_amxx_i386.so    v1.1.1                       pl2  ANY   ANY  
 [15] GeoIP           RUN   -    geoip_amxx_i386.so        v1.10.0.5461                 pl2  ANY   ANY  
15 plugins, 15 running
85.107.127.58:54781:reconnect
amx_modules
Currently loaded modules:
name                    version     author               status     
CURL AMXX               1.1.1       Polarhigh            running    
Orpheu                  2.6.3       joaquimandrade & Ar  running    
ReAPI                   5.22.0.254  Asmodai & s1lent     running    
Engine                  1.10.0.546  AMX Mod X Dev Team   running    
FakeMeta                1.10.0.546  AMX Mod X Dev Team   running    
Ham Sandwich            1.10.0.546  AMX Mod X Dev Team   running    
Fun                     1.10.0.546  AMX Mod X Dev Team   running    
HL Weapon Mod           0.1.0-dev.  AGHL.RU Dev Team     running    
Amxx Easy Http          1.1.1       Polarhigh            running    
GeoIP                   1.10.0.546  AMX Mod X Dev Team   running    
unknown                 unknown     unknown              bad load   
JSON                    1.10.0.546  AMX Mod X Dev Team   running    
RegEx                   1.10.0.546  AMX Mod X Dev Team   running    
13 modules

**Comment by andreiseverin** [comment]
LE: something feels off on loading weaponmod
```L 12/17/2023 - 19:23:16: -------- Mapchange to crossfire --------
[WEAPONMOD]: Start.
[WEAPONMOD]:  Version 0.1.0-dev.1+bc87b53.master Linux
[WEAPONMOD]:   Found (null) at 0xf72f3000
[WEAPONMOD]:   Found (null) at 0xf2b49000
[WEAPONMOD]:   Found hl.so at 0xf29ae000
[WEAPONMOD]:   Found ReHLDS API v3.13
[WEAPONMOD]:  Memory initialization ended.
[WEAPONMOD]: Done.`

**Comment by andreiseverin** [comment]
After testing more times, the problem occurs when changing the map. Similar to [Issue 11](https://github.com/tmp64/weaponmod/issues/11)

**Comment by tmp64** [comment]
I fixed #11. See if it works now

**Comment by andreiseverin** [comment]
@tmp64 it works . Thank you for your help! I'm a big fan of your work!
